Maple2.File.IO/Nif/Endian.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System.Buffers.Binary;
+using System.Numerics;
+using System.Text;
+
+namespace Maple2.File.IO.Nif;
+
+public class EndianReader : BinaryReader {
+    public bool Swap { get; init; }
+
+    public EndianReader(Stream input, bool swap, int index = 0) : base(input) {
+        Swap = swap;
+    }
+
+    public ushort ReadAdjustedUInt16() {
+        ushort value = ReadUInt16();
+
+        if (Swap) {
+            return BinaryPrimitives.ReverseEndianness(value);
+        }
+
+        return value;
```

Comment:
more of a stylistic choice but what about simplifying these to:
```cs
return Swap ? BinaryPrimitives.ReverseEndianness(value) : value;
```

would cut down on this file a lot